lib/linter/linter.js

```diff
@@ -393,7 +393,9 @@ function getDirectiveComments(filename, ast, ruleMapper, warnInlineConfig) {
                         }
 
                         try {
-                            validator.validateRuleOptions(rule, name, ruleValue);
+                            const cloneRuleValue = lodash.cloneDeep(ruleValue);
```

Comment:
Since this configuration exists in a comment that is parsed, do we need to do this? It doesn't seem like it we'll ever be in danger of modifying anything by reference.

Reply:
if we are cloning in the CLI scope (either of these https://github.com/eslint/eslint/pull/13034#discussion_r413592713) then I  think no need to cloning here or anywhere which is currently done in this PR.  
So as you mentioned (https://github.com/eslint/eslint/pull/13034#discussion_r413486744) it would be better to clone while reading the config .